### PR TITLE
feat: expose join_paths as `vim.fs.joinpath`

### PR DESCRIPTION
**Pull request by clason**
This is a small function but used a lot in some plugins.

### CODE REVIEW

**Comment by lewis6991** [comment]
I do think this needs `path` in the name somewhere.

**Comment by gpanders** [comment]
> I do think this needs `path` in the name somewhere. Eventually we'll want `abspath`, `relpath` etc

That's a good point. I'm ok with `joinpath` or `joinpaths`. In either case, I'd prefer we stick to `singleword` style (so e.g. `joinpath` over `join_path`), see a similar discussion in the vim.iter PR https://github.com/neovim/neovim/pull/23029#discussion_r1164601916.

**Review by mfussenegger — APPROVED** [review]

**Comment by clason** [comment]
added a small test @gpanders

**Comment by justinmk** [comment]
> needs `path` in the name somewhere. Eventually we'll want `abspath`, `relpath` etc

`fs.abs` and `fs.rel` seem like they would work fine. I expect the "import" pattern to be `local fs = vim.fs` rather than `local joinpath = vim.fs.joinpath`.

File: test/functional/lua/fs_spec.lua [hunk 1]
```diff
@@ -266,6 +266,17 @@ describe('vim.fs', function()
     end)
   end)
 
+  describe('joinpath()', function()
+    it('works', function()
+      eq('foo/bar/baz', exec_lua([[
+        return vim.fs.joinpath('foo', 'bar', 'baz')
+      ]], nvim_dir))
+      eq('foo/bar/baz', exec_lua([[
+        return vim.fs.joinpath('foo', '/bar/', '/baz')
```
Comment (justinmk) [inline test/functional/lua/fs_spec.lua:275]: Most path join implementations would return `/baz` for this input. E.g. https://docs.python.org/3/library/os.path.html#os.path.join : 

> If a segment is an absolute path (which on Windows requires both a drive and a root), then all previous segments are ignored and joining continues from the absolute path segment.

**Review by justinmk — APPROVED** [review]

**Comment by justinmk** [comment]
> I do think this needs `path` in the name somewhere. Eventually we'll want `abspath`, `relpath` etc

Rethinking this, not sure we need "path" in any of those names. 

We will likely need `vim.uri.join()`, and appending "uri" there seems clumsy?

**Comment by lewis6991** [comment]
> Rethinking this, not sure we need "path" in any of those names.

Then we need `vim.path.*`.

Concise function names are fine but they need to be good names and not ambiguous, especially not for the sake of 4 characters.

**Comment by justinmk** [comment]
Let's keep the current approach then. Almost everything in `vim.fs` is path-related (though I could see us needing `vim.fs.write` et al. in the future). I don't think `vim.fs.rel()` or `vim.fs.abs()` is too bad, but I can live with current approach.

**Pull request merged**